pkgs/dart_mcp_server/tool/update_readme.dart

```diff
@@ -4,51 +4,63 @@
 
 import 'dart:io';
 
-import 'package:dart_mcp/server.dart';
-import 'package:dart_mcp_server/src/mixins/analyzer.dart';
-import 'package:dart_mcp_server/src/mixins/dash_cli.dart';
-import 'package:dart_mcp_server/src/mixins/dtd.dart';
-import 'package:dart_mcp_server/src/mixins/pub.dart';
-import 'package:dart_mcp_server/src/mixins/pub_dev_search.dart';
-
-final Map<String, List<Tool>> toolCategories = {
-  'project': DashCliSupport.allTools,
-  'analysis': DartAnalyzerSupport.allTools,
-  'runtime': DartToolingDaemonSupport.allTools,
-  'pub': PubSupport.allTools,
-  'pub.dev': PubDevSupport.allTools,
-};
-
-void main(List<String> args) {
-  final buf = StringBuffer();
-
-  for (final entry in toolCategories.entries) {
-    final category = entry.key;
-    final tools = entry.value;
-
-    buf.writeln('### $category');
-    buf.writeln('');
-
-    for (final tool in tools) {
-      buf.writeln('- `${tool.name}`: ${tool.description}');
-    }
-
-    buf.writeln('');
+import 'package:dart_mcp/client.dart';
+
+void main(List<String> args) async {
+  print('Getting registered tools...');
+
+  final tools = await _retrieveRegisteredTools();
+
+  final buf = StringBuffer('''
+| Name | Title | Description |
+| --- | --- | --- |
+''');
+  for (final tool in tools) {
+    buf.writeln('| `${tool.name}` | ${tool.title} | ${tool.description} |');
   }
 
   final readmeFile = File('README.md');
-  final updated = insertBetween(
+  final updated = _insertBetween(
     readmeFile.readAsStringSync(),
     buf.toString(),
     '<!-- generated -->',
   );
   readmeFile.writeAsStringSync(updated);
+
+  print('Wrote update tool list to ${readmeFile.path}.');
 }
 
-String insertBetween(String original, String insertion, String marker) {
+String _insertBetween(String original, String insertion, String marker) {
   final startIndex = original.indexOf(marker) + marker.length;
   final endIndex = original.lastIndexOf(marker);
 
   return '${original.substring(0, startIndex)}\n\n'
-      '$insertion${original.substring(endIndex)}';
+      '$insertion\n${original.substring(endIndex)}';
+}
+
+Future<List<Tool>> _retrieveRegisteredTools() async {
+  final client = MCPClient(
```

Comment:
Fwiw, the tools registered can vary based on the client features that are supported.

Probably, this client should claim to support all features so that all the tools show up.

Reply:
I looked at creating a subclass of an `MCPClient` in order to override `initialize()` so I could create specific client capabilities, but for whatever reason, two of the tools dropped out of the list of supported tools. For now, I kept the implementation as is - not providing an explcit list of capabilities.

Reply:
What likely happened there is we swapped out some tools for their preferred alternatives (resources), due to the client indicating it has resource support.

We can land this as is though and I can eventually take a look at improving it.